### PR TITLE
Fix NUTS

### PR DESCRIPTION
**Pull request by rlouf**
_Update:_ The `has_subtree_terminated` condition was not used to stop the tree expansion which made the sampler biased. I corrected this (and simplified the sampling tests). 

- [x] currently the algorithm only runs the integrator once. This is likely due to a problem initializing the termination state.
- [x] We are having similar shape issues in `termination.py` to the one we had in `metrics.py`

### CODE REVIEW

**Comment by rlouf** [comment]
I ended up doing a step-by-step comparison with `blackjax` and it turns out there is one bug left in `termination.py` in the `is_iterative_turning` function. The function should return `False` when `idx_max < idx_min` but is currently returning `True` instead.

I am working on a dirty fix, but the ideal would be to address [this issue in aesara](https://github.com/aesara-devs/aesara/issues/659).

**Comment by rlouf** [comment]
Ah another instance of a test that fails when running `pytest` in the root folder, but passes when I do `pytest tests/test_hmc.py`. There's clearly something I don't understand about `pytest`.

**Comment by brandonwillard** [comment]
> Ah another instance of a test that fails when running `pytest` in the root folder, but passes when I do `pytest tests/test_hmc.py`. There's clearly something I don't understand about `pytest`.

That could be due to dependencies between the RNG seeding and the test-running process.  For example, if tests alter a shared/global RNG state, then they can fail when run in a given order.

File: tests/test_hmc.py [hunk 1]
```diff
@@ -1,108 +1,222 @@
...
+    assert np.mean(samples[1000:]) == pytest.approx(1.0, rel=1e-1)
+    assert np.var(samples[1000:]) == pytest.approx(4.0, rel=1e-1)
+
+
+def test_nuts():
+    """Test the NUTS kernel on a gaussian target."""
+    step_size = 1.0
+    inverse_mass_matrix = at.as_tensor(1.0)
+
+    Y_rv = at.random.normal(1, 2)
```
Comment (brandonwillard) [inline tests/test_hmc.py:66]: Random draws in tests need to be seeded for consistency.

**Review by brandonwillard — CHANGES_REQUESTED** [review]
- When `test_hmc.test_nuts_mcse` is run, there are warning messages that say 
  ```python
  WARNING  aesara.scan.basic:basic.py:749 When the number of steps is fixed and equal to 1, 
  the provided stopping condition, Elemwise{eq,no_inplace}.0 is ignored
  ```
- The first commit title has an unnecessary "WIP -" prefix.

**Pull request merged**